src/ContainerAppHelper.ts

```diff
@@ -0,0 +1,440 @@
+        yamlConfigPath: string) {
+        core.debug(`Attempting to update Container App with name "${containerAppName}" in resource group "${resourceGroup}" from provided YAML "${yamlConfigPath}"`);
+        try {
+            let command = `containerapp update -n ${containerAppName} -g ${resourceGroup} --yaml ${yamlConfigPath}`;
+            await new Utility().executeAndthrowIfError(`az`, command.split(' '));
+        } catch (err) {
+            core.error(err.message);
+            throw err;
+        }
+    }
+
+    /**
+     * Determines if the provided Container App exists in the provided resource group.
+     * @param containerAppName - the name of the Container App
+     * @param resourceGroup - the resource group that the Container App is found in
+     * @returns true if the Container App exists, false otherwise
+     */
+    public async doesContainerAppExist(containerAppName: string, resourceGroup: string): Promise<boolean> {
+        core.debug(`Attempting to determine if Container App with name "${containerAppName}" exists in resource group "${resourceGroup}"`);
+        try {
+            let command = `containerapp show -n ${containerAppName} -g ${resourceGroup} -o none`;
+            const exitCode = await exec.exec(`az`, command.split(' '));
+            return exitCode === 0;
+        } catch (err) {
+            core.warning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Determines if the provided Container App Environment exists in the provided resource group.
+     * @param containerAppEnvironment - the name of the Container App Environment
+     * @param resourceGroup - the resource group that the Container App Environment is found in
+     * @returns true if the Container App Environment exists, false otherwise
+     */
+    public async doesContainerAppEnvironmentExist(containerAppEnvironment: string, resourceGroup: string): Promise<boolean> {
+        core.debug(`Attempting to determine if Container App Environment with name "${containerAppEnvironment}" exists in resource group "${resourceGroup}"`);
+        try {
+            const command = `containerapp env show -n ${containerAppEnvironment} -g ${resourceGroup} -o none`;
+            const exitCode = await exec.exec(`az`, command.split(' '));
+            return exitCode === 0;
+        } catch (err) {
+            core.warning(err.message);
+            return false;
+        }
+    }
+
+    /**
+     * Determines if the provided resource group exists.
+     * @param resourceGroup - the name of the resource group
+     * @returns true if the resource group exists, false otherwise
+     */
+    public async doesResourceGroupExist(resourceGroup: string): Promise<boolean> {
+        core.debug(`Attempting to determine if resource group "${resourceGroup}" exists`);
+        try {
+            const command = `group show -n ${resourceGroup} -o none`;
+            const exitCode = await exec.exec(`az`, command.split(' '));
```

Comment:
Calling `exec` instead of `executeAndthrowIfError` to make sure pipeline does not fail and still is able to complete with a warning.